### ISSUE TITLE
PermissionStatus.undetermined removed - not sure how to work around it.

### ISSUE THREAD

**Issue by m-j-g** [Baseflow/flutter-permission-handler#475]
## 💬 Questions and Help

I noticed in permission_handler: ^6.0.1 that PermissionStatus.undetermined was removed 9 days ago.  I'm trying to figure out how we find out if the user hasn't accepted or denied a permission yet?  Most apps will present a "pre-permission" permission screen before they hit the user with the OS permission screen.  In order to do this, the app needs to know if the user hasn't acted on a permission yet (aka PermissionStatus.undetermined).  

Just looking for a little guidance on how to know if the user hasn't rejected/accepted a permission before.

**Comment by mvanbeusekom** [comment]
That is correct since version 6.0.0 the `PermissionStatus.undetermined` permission has been removed since it is causing difficulties on Android API 30+. 

The permission_handler now follows the same flow as suggested by the Android guidelines. This basically entails the following on Android:

1. Permissions are implicitly `denied` until the user explicitly `granted` them; 
2. On Android API 30+ permissions are `permanentlyDenied` when the user denied permissions two times in a row (pre-API 30 permissions are `permanentlyDenied` when the use selected the "Don't ask again" checkbox);
3. Once the user has denied permissions for the first time the `shouldShowRequestPermissionRationale` method will return `true` indicating the App should show an explanation on why the App requires permission.

On iOS the flow is a bit simpler:

1. Permissions are initially `denied`, this means you can and should request permissions;
2. If the user grants permissions, the `PermissionStatus.granted` is returned;
3. If the user denies permissions, the `PermissionStatus.permanentlyDenied` is returned.

In short the following is true on both platforms:

- When the current status equals `PermissionStatus.denied` you should always request permissions;
- When the current status equals `PermissionStatus.permanentlyDenied` you cannot request permission and you should redirect users to update permissions in the settings of the operating system.

More information on Android permissions can be found here:
- [Request app permissions](https://developer.android.com/training/permissions/requesting)
